orangecontrib/geo/widgets/owscatterplotmap.py

```diff
@@ -0,0 +1,712 @@
+        )
+        request.setAttribute(QNetworkRequest.FollowRedirectsAttribute, True)
+        request.setMaximumRedirectsAllowed(5)
+
+        # Future yielding a QNetworkReply when finished.
+        reply = self._netmanager.get(request)
+        future._reply = reply
+
+        @future.add_done_callback
+        def abort_on_cancel(f):
+            # abort the network request on future.cancel()
+            if f.cancelled() and f._reply is not None:
+                f._reply.abort()
+
+        def on_reply_ready(reply, future):
+            # type: (QNetworkReply, Future) -> None
+
+            # schedule deferred delete to ensure the reply is closed
+            # otherwise we will leak file/socket descriptors
+            reply.deleteLater()
+            future._reply = None
+
+            with closing(reply):
+                if not future.set_running_or_notify_cancel():
+                    return
+
+                if reply.error() == QNetworkReply.OperationCanceledError:
+                    # The network request was cancelled
+                    future.set_exception(Exception(reply.errorString()))
+                    return
+
+                if reply.error() != QNetworkReply.NoError:
+                    # XXX Maybe convert the error into standard http and
+                    # urllib exceptions.
+                    future.set_exception(Exception(reply.errorString()))
+                    return
+
+                try:
+                    image = Image.open(BytesIO(reply.readAll()))
+                except Exception as e:
+                    future.set_exception(e)
+                else:
+                    future.set_result((t, image))
+
+        reply.finished.connect(partial(on_reply_ready, reply, future))
+        return future
+
+
+class OWScatterPlotMap(OWDataProjectionWidget):
+    """Scatter plot visualization of coordinates data with geographic maps for
+    background. Some functionality is copied from OWScatterPlot"""
+
+    name = 'Scatter Plot Geo Map'
+    description = 'Show data points on a world map.'
+    icon = "icons/GeoMap.svg"
+    priority = 101
+
+    tile_provider = settings.Setting(next(iter(TILE_PROVIDERS)))
+    attr_lat = settings.ContextSetting(None)
+    attr_lon = settings.ContextSetting(None)
+    tooltip_shows_all = settings.Setting(True)
+
+    GRAPH_CLASS = OWScatterPlotMapGraph
+    graph = settings.SettingProvider(OWScatterPlotMapGraph)
+    embedding_variables_names = None
+
+    class Warning(OWDataProjectionWidget.Warning):
+        missing_coords = Msg(
+            "Plot cannot be displayed because '{}' or '{}' "
+            "is missing for all data points")
+        no_lat_lon_vars = Msg("Data has no Latitude and Longitude variables")
+        out_of_range = Msg("Plot cannot displayed {} points because "
+                           "Latitude or Longitude values are out of range")
+        large_sql = Msg("Geo Map does not support large Sql tables")
```

Comment:
I would use this for any kind of "big data" --> "Too many data instances to display."